src/components/MinimalDemoWithWeather.js

```diff
@@ -1,34 +1,44 @@
 /**
- * WordPress dependencies
- */
-
-/**
- * Internal dependencies
+ * External dependencies
  */
 import {
 	ChatHistory,
 	MessageContent,
 	PopUpControls,
 	useAgent,
 	useAgentExecutor,
+	useAgents,
 	useChat,
 	UserMessageInput,
 } from '@automattic/big-sky-agents';
+import { useEffect } from 'react';
+
+/**
+ * Internal dependencies
+ */
+
 import withDemoChat from './withDemoChat';
 
 const MinimalDemoUI = () => {
+	const { setActiveAgent, activeAgent, setAgentStarted } = useAgents();
+
 	useAgent(MinimalAgent);
+
+	useEffect(() => {
+		if (activeAgent?.id !== MinimalAgent.id) {
+			setAgentStarted(false);
```

Comment:
@gravityrail because of this[ line here](https://github.com/Automattic/big-sky-agents/blob/add/assistant-model-exports/src/hooks/use-agent-executor.js#L355) the `onStart` is not called when switching tabs, setting it to false here fixes this, but still broken as [toolkitsLoaded](https://github.com/Automattic/big-sky-agents/blob/add/assistant-model-exports/src/hooks/use-agent-executor.js#L351) is false as the weather toolkit is not being registered, try a few different things but nothing worked